generators/chipyard/src/main/scala/TestHarness.scala

```diff
@@ -26,5 +29,32 @@ class TestHarness(implicit val p: Parameters) extends Module {
 
   val dut = p(BuildTop)(p)
   io.success := false.B
-  p(IOBinders).values.map(fn => fn(clock, reset.asBool, io.success, dut))
+  // resetOverride can be overridden via a harnessFunction
+  val resetOverride = Wire(Bool())
+  resetOverride := reset
+  dut.harnessFunctions.foreach(_(this))
+
+  // Aliases for clock, reset, and success
+  def c  = clock
+  def r  = reset.asBool
+  def ro = resetOverride
+  def s  = io.success
+}
+
+object TestHarnessUtils {
+
```

Comment:
Why are these two functions defined here, instead of inlined within the IOBinders? Are they ever reused across multiple IOBinders?

Reply:
They were previously. We can revert it back, but I could see a situation where we do want to split it out in the future. No strong opinions; it's easy to move either way.

Reply:
I went ahead and changed this back.